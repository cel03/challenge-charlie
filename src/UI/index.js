export { Container } from './Container'
export { Template } from './Template'
export { Text } from './Text'
