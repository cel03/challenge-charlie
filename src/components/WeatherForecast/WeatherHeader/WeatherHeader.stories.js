import React from 'react'
import { WeatherHeader } from '.'

export default {
  title: 'WeatherHeader',
  args: {
    cityName: 'Rio de Janeiro',
    stateName: 'Rio de Janeiro',
    onCityInputChange: () => {},
  },
}

const Template = (args) => <WeatherHeader {...args} />

export const Default = Template.bind({})
Default.storyName = 'Default'
