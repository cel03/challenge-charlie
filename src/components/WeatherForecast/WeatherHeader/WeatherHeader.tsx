import React from 'react'
import { RiCompassLine } from 'react-icons/ri'
import { IconContext } from 'react-icons'

import {
  WeatherHeaderContainer,
  WeatherHeaderIcon,
  WeatherHeaderText,
} from './WeatherHeader.styles'

interface WeatherHeaderProps {
  city: string
  state: string
}

const WeatherHeader = ({ city, state }: WeatherHeaderProps) => {
  return (
    <WeatherHeaderContainer>
      <WeatherHeaderIcon>
        <RiCompassLine />
      </WeatherHeaderIcon>
      <WeatherHeaderText>
        {city}, {state}
      </WeatherHeaderText>
    </WeatherHeaderContainer>
  )
}

const WeatherHeaderWrapper = (props: WeatherHeaderProps) => (
  <IconContext.Provider value={{ size: '35' }}>
    <WeatherHeader {...props} />
  </IconContext.Provider>
)
export default WeatherHeaderWrapper
