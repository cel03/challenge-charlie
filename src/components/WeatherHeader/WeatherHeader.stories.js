import React from 'react'
import { WeatherHeader } from './'

export default {
  title: 'WeatherHeader',
  args: {
    cityName: 'Rio de Janeiro',
    stateName: 'Rio de Janeiro',
  },
}

const Wrapper = ({ children }) => (
  <div style={{ maxWidth: 300 }}>{children}</div>
)
const Template = (args) => (
  <Wrapper>
    <WeatherHeader {...args} />
  </Wrapper>
)

export const Default = Template.bind({})
Default.storyName = 'Default'
