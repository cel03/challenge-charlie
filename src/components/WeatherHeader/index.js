export { default as WeatherHeader } from './WeatherHeader'
