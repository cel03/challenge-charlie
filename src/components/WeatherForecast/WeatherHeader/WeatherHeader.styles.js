import styled from 'styled-components'
import { Container } from '../../../UI'

export const WeatherHeaderContainer = styled(Container)`
  color: rgb(126, 126, 124);
  background-color: rgba(243, 238, 236, 0.92);
  height: initial;
`

export const WeatherHeaderIcon = styled.div`
  width: auto;
  margin-right: 8px;
`

export const WeatherHeaderText = styled.span`
  width: 100%;
  text-align: start;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`
