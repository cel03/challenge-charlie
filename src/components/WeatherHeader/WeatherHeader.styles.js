import styled from 'styled-components'
import { Container } from '../../UI/'

export const WeatherHeaderContainer = styled(Container)`
  color: rgb(126, 126, 124);
  background-color: rgba(243, 238, 236, 0.5);
  height: initial;
`

export const WeatherHeaderIcon = styled.div`
  min-width: 50px;
`

export const WeatherHeaderText = styled.div`
  width: 100%;
  text-align: start;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`
