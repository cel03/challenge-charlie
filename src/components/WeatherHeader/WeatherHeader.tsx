import React from 'react'
import { RiCompassLine } from 'react-icons/ri'
import { IconContext } from "react-icons";

import { WeatherHeaderContainer, WeatherHeaderIcon, WeatherHeaderText } from './WeatherHeader.styles'

interface WeatherHeaderProps {
    cityName: string,
    stateName: string
}

const WeatherHeader = ({ cityName, stateName }: WeatherHeaderProps) => {
    return (
        <WeatherHeaderContainer >
            <WeatherHeaderIcon>
                <RiCompassLine />
            </WeatherHeaderIcon>
            <WeatherHeaderText>
                {cityName}, {stateName}
            </WeatherHeaderText>
        </WeatherHeaderContainer>
    )
}

const WeatherHeaderWrapper = (props: WeatherHeaderProps) => (
    <IconContext.Provider value={{ size: '35' }}>
        <WeatherHeader {...props} />
    </IconContext.Provider >
)
export default WeatherHeaderWrapper