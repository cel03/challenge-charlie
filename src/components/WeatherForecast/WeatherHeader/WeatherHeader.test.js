import React from 'react'
import { render } from '@testing-library/react'
import { Template } from '../../UI'
import WeatherHeader from './WeatherHeader'

describe('WeatherHeader', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <Template>
        <WeatherHeader cityName="Rio de Janeiro" stateName="Rio de Janeiro" />
      </Template>
    )
    expect(container).toMatchSnapshot()
  })
})
