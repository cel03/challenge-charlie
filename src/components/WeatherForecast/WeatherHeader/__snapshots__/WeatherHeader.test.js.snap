// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherHeader matches snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL sc-dkPtRN dzyvjz ffyhVb"
  >
    <div
      class="sc-hKwDye jxGrwj"
    >
      <svg
        fill="currentColor"
        height="35"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm3.5-11.5l-2 5-5 2 2-5 5-2z"
          />
        </g>
      </svg>
    </div>
    <div
      class="_1EEDX _3sXmF _3qwMT sc-eCImPb capuZz"
    >
      Informe sua cidade
    </div>
  </div>
</div>
`;
